functorch/_src/aot_autograd.py

```diff
@@ -765,20 +783,20 @@ def get_aot_graph_name() -> str:
     Returns the name of the graph being compiled.
     """
     global model_name, graph_being_compiled, nth_graph
-    return f"{model_name}_{'_'.join(graph_being_compiled)}_{nth_graph}"
+    return f"{model_name}__{'_'.join(graph_being_compiled)}_{nth_graph}"
 
 
 get_graph_being_compiled = get_aot_graph_name
 
 
 @contextmanager
-def track_graph_compiling(graph_name, increment_index=False):
+def track_graph_compiling(aot_config, graph_name):
     global graph_being_compiled
-    graph_being_compiled = [graph_name]
+    # TODO: Don't shove the aot_id in here; set it in the context
+    graph_being_compiled = [f"{aot_config.aot_id}_{graph_name}"]
     yield
-    if increment_index:
```

Comment:
It looks like we call `track_graph_compiling` both when we call `partition_fn()`, **and** when we call `fw_compiler()`. It seems like that would mean `nth_graph` gets a different value at each invocation, is that what we want?

Reply:
I think the original logic was trying to only increment on backwards, so that you would get a distinct nth_graph number per aot autograd call. But this logic was wrong: because backwards is not compiled eagerly anymore, you would end up with a bunch of forward graphs allocated the same number, and then all the backward compiles  increment all at once.

Maybe @Chillee knows more about what the intended semantics are, but the easiest fix was to just make nth_graph globally unique, and rely on aot_id for correlating otherwise.